todoapp/src/components/TodoList.vue

```diff
@@ -2,44 +2,65 @@
 import { onMounted, ref, watch } from 'vue'
 import moment from 'moment'
 import type { TodoType } from '../types/text'
-
+import DeletePopUP from './DeletePopUP.vue'
 import Header from './Header.vue'
 import Todos from './Todos.vue'
 
 const todos = ref<TodoType[]>([])
+const popUp = ref(false)
+const selectedTaskIndex = ref()
 
 function addTodo() {
   const currentDate = moment().format('DD.MM.YYYY')
   const newTodo: TodoType = {
-    title: 'Pay for rent',
-    text: 'Lorem ipsum dolor sit amet ipsum dolor sit amet',
+    title: 'Add a title',
+    text: 'Add a description',
     priority: 0,
+    priorityChange: false,
     status: false,
-    created_at: ` ${currentDate}`
+    created_at: ` ${currentDate}`,
+    editing: false
   }
   todos.value.push(newTodo)
 }
-
 onMounted(() => {
   todos.value = JSON.parse(localStorage.getItem('todos')!) || []
 })
-
 watch(
   todos,
   (newVal) => {
     localStorage.setItem('todos', JSON.stringify(newVal))
   },
   { deep: true }
 )
-</script>
 
+function deleteTaskIndex(index: number) {
+  popUp.value = true
+  selectedTaskIndex.value = index
+}
+
+function togglePopUP() {
+  popUp.value = !popUp.value
+}
+function removeTask(index: number) {
+  todos.value.splice(index, 1)
+  togglePopUP()
+}
+</script>
 <template>
+  <div class="delete-popup">
+    <DeletePopUP
+      v-if="popUp"
+      @togglePopUP="togglePopUP"
```

Comment:
Please use hyphenated format for all your props and events and respect the order of attributes props goes first and the emit ( move `:index="selectedTaskIndex"` before your events)